inetdiag/inetdiag_test.go

```diff
@@ -290,4 +292,38 @@ func TestReader(t *testing.T) {
 	}
 }
 
+func TestNLMsgSerialize(t *testing.T) {
+	source := "testdata/testdata.zst"
+	log.Println("Reading messages from", source)
+	rdr := zstd.NewReader(source)
+	parsed := 0
+	for {
+		msg, err := inetdiag.LoadNext(rdr)
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			t.Fatal(err)
+		}
+		pm, err := inetdiag.Parse(msg, false)
+		rtx.Must(err, "Could not parse test data")
+		s, err := json.Marshal(pm)
+		rtx.Must(err, "Could not serialize %v", pm)
+		if strings.Contains(string(s), "\n") {
+			t.Errorf("String %q should not have a newline in it", s)
+		}
+		var um inetdiag.ParsedMessage
+		rtx.Must(json.Unmarshal([]byte(s), &um), "Could not parse one line of output")
+		rs, err := json.Marshal(um)
+		rtx.Must(err, "Could not remarshall %v", um)
+		if string(rs) != string(s) {
```

Comment:
Add comment:
```go
// Must test double marshalling because variable pm is not DeepEqual to um, because the serialization of attributes does not encode the index number
```

Reply:
Actually, don;t double-encode. Instead, zero out all the non-serialized sub fields and check with DeepEquals